modules/Microsoft.EventHub/namespaces/networkRuleSets/deploy.bicep

```diff
@@ -5,32 +5,39 @@ param namespaceName string
   'Enabled'
   'Disabled'
 ])
-@description('Optional. This determines if traffic is allowed over public network. Default it is "Enabled". If set to "Disabled", traffic to this namespace will be restricted over Private Endpoints only.')
+@description('Optional. This determines if traffic is allowed over public network. Default is "Enabled". If set to "Disabled", traffic to this namespace will be restricted over Private Endpoints only.')
 param publicNetworkAccess string = 'Enabled'
 
 @allowed([
   'Allow'
   'Deny'
 ])
-@description('Optional. Default Action for Network Rule Set. Default is "Allow". Will be set to "Deny" if ipRules/virtualNetworkRules or are being used. If ipRules/virtualNetworkRules are not used and PublicNetworkAccess is set to "Disabled", setting this to "Deny" would render the namespace resources inaccessible for data-plane requests.')
```

Comment:
If public network access is set to disabled, traffic is restricted over private endpoint and network rules are not applied. I'm actually thinking about updating the default action as follows (need to test it), if you agree:

`defaultAction: publicNetworkAccess == 'Disabled' ? null : (!empty(ipRules) || !empty(virtualNetworkRules) ? 'Deny' : defaultAction)`

In that way the defaultAction won't be applied at all if the public access is disabled, similarly to iprules and vnetrules, as it should be.

P.S. I didn't find that sentence in the UI, could you please point me to that?

Reply:
@eriqua I can't seem to find the statement in the UI anymore, or maybe it wasn't in the UI but the actual API. Anyways I managed to reproduce it as well with different combinations:

![image](https://user-images.githubusercontent.com/28486158/180378096-e552b496-f714-48bd-b8a4-e3af1f2c1767.png)

I also like your idea. I think it'll protect from this error as well

Reply:
Thanks @ahmadabdalla very useful as always :) 

- I updated the implementation to be led by the publicNetworkAccess parameter. This would be similar to what the portal does afterall, if public access is disabled, network rules are not applied. 
- I updated the parameter descriptions to reflect the above.
- I updated the condition from the parent module
- Implemented the above for both eventhub and servicebus

Reply:
@ahmadabdalla I've updated issue #819 based on what discussed above. Please have a look and feel free to comment on it if you see that something is missing

Reply:
Thank you @eriqua